website/docs/flow/context/index.md

```diff
@@ -0,0 +1,129 @@
+---
+title: Flow Context
+---
+
+Each flow execution has an independent context. This context can hold arbitrary data, which can be used and transformed by stages and policies.
+
+Any data can be stored in the flow context, however there are some reserved keys in the context dictionary that are used by authentik stages.
+
+## Common objects
+
+### `pending_user` (User object)
+
+`pending_user` is used by multiple stages. In the context of most flow executions, it holds the user that is executing the flow. This value is not set automatically, it is set via the [Identification stage](../stages/identification/).
+
+Stages that require a user, such as the [Password stage](../stages/password/), the [Authenticator validation stage](../stages/authenticator_validate/) and others will use this value if it is set, and fallback to the request's users when possible.
+
+### `prompt_data` (Dictionary)
+
+`prompt_data` is primarily used by the [Prompt stage](../stages/prompt/). The value of any field within a prompt stage is written to the `prompt_data` dictionary. For example, given a field with the _Field key_ `email` that was submitted with the value `foo@bar.baz` will result in the following context:
+
+```json
+{
+    "prompt_data": {
+        "email": "foo@bar.baz"
+    }
+}
+```
+
+This data can be modified with policies. The data is also used by stages like [User write](../stages/user_write.md), which takes data in `prompt_data` and writes it to `pending_user`.
+
+### `redirect` (string)
+
+Stores the final redirect that the user's browser will be sent to after the flow is finished executing successfully. This is set when an un-authenticated user attempts to access a secured application, and when a user authenticates/enrolls with an external source.
+
+### `application` (Application object)
+
+When an unauthenticated user attempts to access a secured resource, they are redirected to an authentication flow. The application they attempted to access will be stored in this key.
+
+### `source` (Source object)
+
+When a user authenticates/enrolls via an external source, this will be set to the source they are using.
```

Comment:
Will there be a key attached to the source object? Like `source.SAML`

What do we mean by "source" here, and in general?

Reply:
same as above

Sources are https://goauthentik.io/docs/terminology#source, i.e. federation sources, like log into authentik with github